src/csharp/Grpc.Tools/build/_protobuf/Google.Protobuf.Tools.targets

```diff
@@ -115,27 +116,35 @@
           DependsOnTargets="Protobuf_Compile"
           Condition=" '$(Language)' == 'C#' " />
 
-  <Target Name="_Protobuf_SelectFiles">
+  <Target Name="_Protobuf_SetProtoRoot">
```

Comment:
Can you explain the logic around Protobuf_ProtoRoot a bit more? I'm not sure I understand the purpose of it.

Reply:
`_Protobuf_SetProtoRoot` is an extract from `_Protobuf_SelectFiles`. `Protobuf_Clean` has an bug in C++ because it's missing the `ProtoRoot`. As this PR would also use `ProtoRoot` for calculating `OutputDir` in C#, this bug would also get copied over to C#. Therefore I fixed it in here instead of in a separate PR.

But we can also put it in a separate PR which blocks this one as it's also a bug by itself.